packages/react/src/tanstackrouter.ts

```diff
@@ -0,0 +1,71 @@
+import { WINDOW, startBrowserTracingNavigationSpan, startBrowserTracingPageLoadSpan } from "@sentry/browser";
+import {
+	SEMANTIC_ATTRIBUTE_SENTRY_OP,
+	SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN,
+	SEMANTIC_ATTRIBUTE_SENTRY_SOURCE,
+} from "@sentry/core";
+
+import { browserTracingIntegration as originalBrowserTracingIntegration } from "@sentry/react";
+import type { Client, Integration } from "@sentry/types";
+
+import type { TanstackRouter } from "./types";
+
+/**
+ * A custom browser tracing integration for Tanstack Router.
+ */
+export function tanstackRouterBrowserTracingIntegration(
+	router: TanstackRouter,
+	options: Parameters<typeof originalBrowserTracingIntegration>[0] = {}
+): Integration {
+	const browserTracingIntegrationInstance = originalBrowserTracingIntegration({
+		...options,
+		instrumentNavigation: false,
+		instrumentPageLoad: false,
+	});
+
+	const { instrumentPageLoad = true, instrumentNavigation = true } = options;
+
+	return {
+		...browserTracingIntegrationInstance,
+		afterAllSetup(client) {
+			const initPathName = WINDOW && WINDOW.location && WINDOW.location.pathname;
+			if (instrumentPageLoad && initPathName) {
+				startBrowserTracingPageLoadSpan(client, {
+					name: initPathName,
```

Comment:
is there no way for us to get the `routeId` for the pageload? We can read this from `router.state` also right?

Reply:
Yes, I believe so. Sorry, I missed putting it there. @lforst can you put that in?

Reply:
- Route ID is on each match. You'll probably want the deepest one: `matches[matches.length - 1].routeId`
- Entire route config is available at: `router.routesById[routeId]`

Reply:
I did the `matches[matches.length - 1].routeId` approach below, just forgot to put it here as well.